### ISSUE TITLE
error messages and hints should be asserted in tests

### ISSUE THREAD

**Issue by magurotuna** [denoland/deno_lint#395]
Current tests for the rules are checking if error spans and the number of errors match what we expect. I'd like error messages and hints to be asserted as well, because there are some rules that have more than one possible messages.

For example, `getter-return` has two possible error messages in [ESLint](https://github.com/eslint/eslint/blob/v7.11.0/lib/rules/getter-return.js#L58-L61) (and `deno_lint` imitates it):

1. `Expected to return a value in {{name}}.`
2. `Expected {{name}} to always return a value.`

So, the design of new assertion I'm thinking now looks like:

```rust
assert_lint_err!(
  GetterReturn,
  [
    {
      code: "const foo = { get getter() {} };",
      col: 14,
      message: "Expected to return a value in getter.",
    },
    {
      code: r#"
class Foo {
  get foo() {
    class Bar {
      get bar() {}
    };
    return 42;
  }
}
      "#,
      line: 5,
      col: 6,
      message: "Expected to return a value in bar.",
      hint: "This is a hint",
    },
  ],
);
```

- `code`, `col` and `message` are required
- `line` can be omitted if an error is at line 1
- `hint` can be omitted if there's no hint

@bartlomieju Any thoughts on this?

**Comment by bartlomieju** [comment]
@magurotuna I agree with your sentiment and it SGTM - I'm not a big fan of macros but in this case it does make sense. Will you look into that?

**Comment by magurotuna** [comment]
@bartlomieju yes, I have been working on this, Maybe I'll open a PR in today.
